gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -859,55 +859,64 @@ async fn test_gateway_configuration() -> anyhow::Result<()> {
 
 #[tokio::test(flavor = "multi_thread")]
 async fn test_gateway_supports_connecting_multiple_federations() -> anyhow::Result<()> {
-    multi_federation_test(LightningNodeType::Lnd, |_, rpc, fed1, fed2, _| async move {
-        assert_eq!(rpc.get_info().await.unwrap().federations.len(), 0);
-
-        let invite1 = fed1.invite_code();
-        let info = rpc
-            .connect_federation(ConnectFedPayload {
-                invite_code: invite1.to_string(),
-            })
-            .await
-            .unwrap();
-
-        assert_eq!(info.federation_id, invite1.id);
-
-        let invite2 = fed2.invite_code();
-        let info = rpc
-            .connect_federation(ConnectFedPayload {
-                invite_code: invite2.to_string(),
-            })
-            .await
-            .unwrap();
-        assert_eq!(info.federation_id, invite2.id);
-        Ok(())
-    })
+    multi_federation_test(
+        LightningNodeType::Lnd,
+        |gateway, rpc, fed1, fed2, _| async move {
+            info!("Starting test_gateway_supports_connecting_multiple_federations");
+            assert_eq!(rpc.get_info().await.unwrap().federations.len(), 0);
+
+            let invite1 = fed1.invite_code();
+            let info = rpc
+                .connect_federation(ConnectFedPayload {
+                    invite_code: invite1.to_string(),
+                })
+                .await
+                .unwrap();
+
+            assert_eq!(info.federation_id, invite1.id);
+
+            let invite2 = fed2.invite_code();
+            let info = rpc
+                .connect_federation(ConnectFedPayload {
+                    invite_code: invite2.to_string(),
+                })
+                .await
+                .unwrap();
+            assert_eq!(info.federation_id, invite2.id);
+            drop(gateway); // keep until the end to avoid the gateway shutting down too early
```

Comment:
Wont `drop` be called automatically after it goes out of scope?

Reply:
Yes, but if we don't use this variable it will drop before the test even starts